__tests__/server/plugins/reactHtml/staticErrorPage.spec.jsx

```diff
@@ -14,32 +14,337 @@
  */
 
-import serverError from '../../../src/server/middleware/serverError';
+import Fastify from 'fastify';
+import staticErrorPage, {
+  setErrorPage,
+} from '../../../../src/server/plugins/reactHtml/staticErrorPage';
 
-// existing coverage in ssrServer.spec.js
+jest.mock('@americanexpress/fetch-enhancers', () => ({
+  createTimeoutFetch: jest.fn(
+    (timeout) => (next) => (url) => next(url)
+      .then((res) => {
+        res.timeout = timeout;
+        return res;
+      })
+  ),
+}));
 
-jest.spyOn(console, 'error').mockImplementation(() => {});
+jest.spyOn(console, 'info').mockImplementation(() => { });
+jest.spyOn(console, 'log').mockImplementation(() => { });
+jest.spyOn(console, 'error').mockImplementation(() => { });
+jest.spyOn(console, 'warn').mockImplementationOnce(() => { });
 
-describe('serverError', () => {
-  it('handles req with no headers however unlikely', () => {
-    const reqWithNoHeaders = {};
-    const res = { status: jest.fn(), send: jest.fn() };
-    const next = jest.fn();
+describe('staticErrorPage', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('sends default static error page', async () => {
+    const app = Fastify();
+
+    app.get('/', (request, reply) => staticErrorPage(request, reply));
+
+    const response = await app.inject({
+      method: 'get',
+      url: '/',
+    });
+
+    expect(response.statusCode).toBe(200);
+    expect(response.body).toContain('<!DOCTYPE html>');
+    expect(response.body).toContain('<meta name="application-name" content="one-app">');
+  });
+
+  it('returns message as json', async () => {
+    const app = Fastify();
+
+    app.get('/', (request, reply) => {
+      request.headers['content-type'] = 'application/json';
+
+      staticErrorPage(request, reply);
+    });
+
+    const response = await app.inject({
+      method: 'get',
+      url: '/',
+    });
+
+    expect(console.error).not.toHaveBeenCalled();
+    expect(response.statusCode).toBe(200);
+    expect(await response.json()).toEqual({ message: 'Sorry, we are unable to load this page at this time. Please try again later.' });
+  });
+
+  it('does not change the status code if already set', async () => {
+    const app = Fastify();
+
+    app.get('/', (request, reply) => {
+      reply.code(400);
+
+      staticErrorPage(request, reply);
+    });
+
+    const response = await app.inject({
+      method: 'get',
+      url: '/',
+    });
+
+    expect(response.statusCode).toBe(400);
+  });
+
+  it('invites the user to try again if the status code is 5xx level', async () => {
+    const app = Fastify();
+
+    app.get('/', (request, reply) => {
+      staticErrorPage(request, reply);
+    });
+
+    const response = await app.inject({
+      method: 'get',
+      url: '/',
+    });
+
+    expect(response.body).toContain('Sorry, we are unable to load this page at this time. Please try again later.');
+  });
+
+  it('invites the user to try again if the status code is 404', async () => {
```

Comment:
Why try again on a 404?

Reply:
the "try again" is related to the message that's being rendered: "Please try again later." (see https://github.com/americanexpress/one-app/blob/8209bbe713d88971a9557ab9f681e23ea6baac4a/__tests__/server/middleware/sendHtml.spec.js#L870)